Soa bot/messaging

kartik2112 opened the pull request:
Added function `get_proposals` to `smart_order_acceptor_bot.py` that would extract proposals from received messages by verifying sentence structure.

Added function `gen_pos_stance_messages` to `smart_order_acceptor_bot.py` that would check for stance vector and accordingly send messages to other powers with positive stance.

trigaten commented on src/baseline_bots/bots/dipnet/rerank_dipnet_bot.py:
```diff
@@ -0,0 +1,162 @@
+import time
+import numpy as np
+
+sys.path.append("..")
```
Please remove all `sys.path.append` statements.

trigaten commented on src/baseline_bots/bots/dipnet/rerank_dipnet_bot.py:
```diff
@@ -0,0 +1,162 @@
+			sim_game.set_orders(power_name=other_power, orders=other_pow_orders[other_power])
+		sim_game.set_orders(power_name=self.power_name, orders=orders)
+		# print()
+		# print("Orders")
```
Remove all commented out code

trigaten commented on src/baseline_bots/bots/smart_order_accepter_bot.py:
```diff
@@ -1,12 +1,15 @@
 from baseline_bots.utils import MessagesData, OrdersData, get_other_powers
+from baseline_bots.stance import stance_extraction
 
+# Using library https://github.com/SHADE-AI/daidepp
```
Please add this library to the `requirements.txt` file

trigaten commented on src/baseline_bots/bots/smart_order_accepter_bot.py:
```diff
@@ -24,6 +27,42 @@ class SmartOrderAccepterBot(RandomProposerBot):
+
+        return proposals
+
+    def gen_pos_stance_messages(self, game_rec, ret_msgs) -> None:
```
Please add type annotations

trigaten commented on src/baseline_bots/bots/dipnet/rerank_dipnet_bot.py:
```diff
@@ -0,0 +1,162 @@
+
+	@gen.coroutine
+	def gen_messages(self, rcvd_messages:List[Message]) -> MessagesData:
+		"""query dipnet for orders"""
```
This docstring seems wrong or perhaps the function is incomplete?

trigaten reviewed (changes requested):
Hey Kartik! This looks great. I just put a few comments which should be easily resolved. What is the point of the `ReRankDipnetBot` bot? I dont see this mentioned in your PR comment, nor used in the new code.

trigaten commented:
I think the CI is failing since SOA bot isnt imported here: https://github.com/ALLAN-DIP/baseline_bots/blob/main/src/baseline_bots/bots/__init__.py

kartik2112 commented:
Removed rerank bot. It is a parallel experiment which I have now moved to a different branch.

Added type annotations to both functions. Addressing issues #26 and #32

trigaten commented on src/baseline_bots/bots/smart_order_accepter_bot.py:
```diff
@@ -24,6 +27,42 @@ class SmartOrderAccepterBot(RandomProposerBot):
+
+        return proposals
+
+    def gen_pos_stance_messages(self, game_rec: json, ret_msgs: MessagesData) -> None:
```
is json a Python type?

kartik2112 replied:
I'll change this to Dict which seems more general and more appropriate

kartik2112 replied:
But I am not sure what would be the type annotation for Dict's values since I had not worked on these types of game records.

trigaten commented:
Would you mind putting your function call(s) in the appropriate location of the SOA bot `__call__` function as shown in the roadmap?

kartik2112 commented:
I can add the straightforward call to `get_proposals()` but I am unaware of the dynamics needed for `get_stance` to work since I am not sure how the scores would be added to the game file. I just added a template function from what I knew. This would need further disucssion.

trigaten commented:
> I can add the straightforward call to `get_proposals()` but I am unaware of the dynamics needed for `get_stance` to work since I am not sure how the scores would be added to the game file. I just added a template function from what I knew. This would need further disucssion.

Fair enough. Instead, can you modify 
`gen_messages(self)` to call your functions and return the results.
For now, it should just return messages containing true orders for positively stanced opponents.
@kkahadze Will fill in the functionality for negative stanced opponents later.

trigaten commented:
@kartik2112 Please resolve the merge conflicts locally

kartik2112 commented:
Addressed #27 as well

trigaten commented on src/baseline_bots/bots/smart_order_accepter_bot.py:
```diff
@@ -24,11 +29,57 @@ class SmartOrderAccepterBot(RandomProposerBot):
+            if self.stance.stance[self.power_name][pow] > 0:
+                msgs_data.add_message(pow, str(orders_decided))
 
     def gen_messages(self, rcvd_messages):
```
Can you modify the signature so that it takes in the order it is going to execute.

it should be the following (add a type hint): `def gen_messages(self, orders):`

Then generate the messages accordingly.

kartik2112 replied:
And take the stance computation out of gen_messages then?

trigaten commented on src/baseline_bots/bots/smart_order_accepter_bot.py:
```diff
@@ -1,12 +1,17 @@
 __email__ = "sanderschulhoff@gmail.com"
 
-from DAIDE import ALY, PRP
+import json
```
Is this used anywhere?

trigaten reviewed (changes requested):
Almost done :)

trigaten reviewed (approved):
This is good, thanks.
